src/validate.ts

```diff
@@ -0,0 +1,122 @@
+'use-strict';
+
+/**
+ * Generates random request ID 
+ * @param min Minimum number
+ * @param max Maximum number 
+ * @returns Random number between the range of the min to max number specified
+ */
+function generateRequestID(min: number, max: number){
+     return Math.floor(Math.random() * (max-min) + min).toString()
+}
+
+/**
+ * Generates SYNC request 
+ * @returns Specified format for SYNC intent request. 
+ */
+export function generateSyncRequest(){
+    const requestId = generateRequestID(100,999);
+    return {
+        requestId,
+        inputs: [{
+          intent: "action.devices.SYNC"
+        }]
+    }
+}
+
+interface Device {
+  id: string[],
+  customData?: {[key: string]: any}
+}
+
+interface QueryRequest{
+    requestId: string;
+    inputs: {
+     intent: string; 
+     payload: {
+         devices: {
+             id: string; 
+             customData: Device["customData"];
+        }[]; 
+     };
+    }[];
+}
+
+/**
+ * Generates QUERY request 
+ * @param deviceIds array of strings specifying the device deviceIds
+ * @param customData an array of data for the respective devices specified by the developer 
+ * @returns Specified format for QUERY intent request. 
+ */
+export function generateQueryRequest(deviceIds: Device["id"],customData: Device["customData"]) : QueryRequest{
```

Comment:
Wouldn't you just provide a `Device[]`? Here we are breaking apart the device just to reassemble it in the request payload.

Reply:
I get an error here since having it set as Device[] affects the devices mapping part in the requests, its no longer an array of strings if I only call Device[]. However, are you suggesting a merge between deviceIds + customData so I'm only taking in one parameter instead of two?

Reply:
Yes, that's why you could create a `Device` type that encapsulates both. Here is a full example of what this might look like for query:

```typescript
interface Device {
  id: string;
  customData?: {[key: string]: any};
}

interface QueryInput {
  intent: string; 
  payload: {
    devices: Device[]; 
  };
}

interface QueryRequest {
  requestId: string;
  inputs: QueryInput[];
}

/**
 * Generates QUERY request 
 * @param deviceIds array of strings specifying the device deviceIds
 * @param customData an array of data for the respective devices specified by the developer 
 * @returns Specified format for QUERY intent request. 
 */
export function generateQueryRequest(devices: Device[]) : QueryRequest {
    const requestId = generateRequestID(100,999);
    return {
      requestId,
      inputs: [{
        intent: "action.devices.QUERY",
        payload: {
          devices,
        }
      }]
    };
}
```

Reply:
Gotcha! Thanks for the clarification - making these edits and doing the same with the execute request function.